kdwsdl2cpp/src/converter_complextype.cpp

```diff
@@ -310,7 +350,7 @@ QString Converter::generateMemberVariable(const QString &rawName, const QString
 
     // getter method
     QString getterTypeName = typeName;
-    if (optional) {
+    if (optional && defaultValue.mValue.isNull()) {
```

Comment:
I suggest to add a `defaultValue.isNull()` method to encapsulate the logic. If one day you realize that "" (isEmpty()) should also be considered as "no default value" then you'll have a single place to change.

Reply:
fixed